### PR TITLE
[2.11] Remove all references to Restricted Admin

### PR DESCRIPTION
**Pull request by JonCrowther**
## Issue: https://github.com/rancher/rancher/issues/47875

Related webhook PR: https://github.com/rancher/webhook/pull/549

## Problem
Restricted Admin has been deprecated since 2.8 and replaceable since v2.9. Now that it has gone through the deprecation cycle, it is ready to be removed from the code base in 2.11.
 
## Solution
Removed all references to Restricted Admin from the code. 
### Manual Testing
Built and checked that the restricted admin cluster roles, bindings and role template were gone. Unit tests are also passing.

### Automated Testing
* Test types added/modified:
    * Unit
    * Validation (Go Framework)

Summary: Removed tests specific to restricted admin since they are no longer applicable.

## QA Testing Considerations
The upgrade scenario is an important one. If a system has existing restricted admin users, they should still exist and not produce errors. However, those users will experience disruptions since all the custom handling of restricted admin has been removed. So some resources they previously accessed may no longer work.
 
### Regressions Considerations
The biggest regression possibility is the upgrade path.

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

**Comment by JonCrowther** [comment]
@anupama2501 I added you as a reviewer because I've removed the Restricted Admin parts of this PR you authored https://github.com/rancher/rancher/pull/41888

**Comment by JonCrowther** [comment]
The only reference to Restricted Admin I left is in this readme: https://github.com/rancher/rancher/blob/main/tests/v2/validation/rbac/globalrolesv2/README.md

I left it because the reference is providing context to what `globalrolesv2` tests are for. I can remove it if anyone feels strongly against leaving it.

File: tests/v2/validation/charts/webhook_test.go [hunk 1]
```diff
@@ -109,27 +108,6 @@ func (w *WebhookTestSuite) TestWebhookChart() {
 	}
 }
 
-func (w *WebhookTestSuite) TestWebhookEscalationCheck() {
```
Comment (anupama2501) [inline tests/v2/validation/charts/webhook_test.go:112]: tagging @Priyashetty17 from collie team as we might have to add something to replace the WebhookescalationCheck

**Review by anupama2501 — APPROVED** [review]
lgtm on the tests

File: pkg/rbac/common.go [hunk 2]
```diff
@@ -20,17 +20,12 @@ import (
...
-	ProjectCRDsClusterRole            = "project-crd-clusterRole"
-	RestrictedAdminProjectRoleBinding = "restricted-admin-rb-project"
-	RestrictedAdminCRForClusters      = "restricted-admin-cr-clusters"
-	RestrictedAdminCRBForClusters     = "restricted-admin-crb-clusters"
+	NamespaceID            = "namespaceId"
+	ProjectID              = "projectId"
+	ClusterID              = "clusterId"
+	GlobalAdmin            = "admin"
+	ClusterCRDsClusterRole = "cluster-crd-clusterRole"
+	ProjectCRDsClusterRole = "project-crd-clusterRole"
```
Comment (raulcabello) [inline pkg/rbac/common.go:28]: `ClusterCRDsClusterRole` and `ProjectCRDsClusterRole` are not used anymore. Can you remove them, please?

**Review by raulcabello — COMMENTED** [review]
Left a small comment about 2 vars not being used anymore. Apart from that, LGTM. I can't find any more references to Restricted Admin in the code.

File: pkg/rbac/common.go (hunk 2 above)
Comment (JonCrowther) [reply pkg/rbac/common.go]: Good catch, removed

**Review by raulcabello — APPROVED** [review]

**Review by pmatseykanets — APPROVED** [review]

**Pull request merged**